### PR TITLE
fix rubocop

### PR DESCRIPTION
**Pull request by aljesusg**
- [X] Helpers Rubocop
- [X] Models
  - [X] Event
  - [X] Charge
  - [X] Pool
  - [X] PricePlan
  - [X] Rate
  - [X] Tier
  - [X] UsageType
  - [X] ConsumptionManager
- [X] Tests
  - [X] Helpers
  - [X] Models

### CODE REVIEW

File: app/helpers/manageiq/consumption/utils_helper.rb [hunk 1]
```diff
@@ -36,11 +37,8 @@ def self.get_parent(resource)
     def self.get_type_hierarchy_next(resource)
       resource_type = resource.type.split("::")[-1] unless resource.type.nil?
       # I get the next type of resource parent
-      case
-      when HARDWARE_RESOURCE.include?(resource_type) then HARDWARE_RESOURCE[HARDWARE_RESOURCE.index(resource_type)+1] || ""
-      when CONTAINER_RESOURCE.include?(resource_type) then CONTAINER_RESOURCE[CONTAINER_RESOURCE.index(resource_type)+1] || ""
-      else ""
-      end
+      if HARDWARE_RESOURCE.include?(resource_type) then HARDWARE_RESOURCE[HARDWARE_RESOURCE.index(resource_type) + 1] || "" end
+      if CONTAINER_RESOURCE.include?(resource_type) then CONTAINER_RESOURCE[CONTAINER_RESOURCE.index(resource_type) + 1] || "" end
```
Comment (Fryguy) [inline app/helpers/manageiq/consumption/utils_helper.rb]: Prefer ternary operator over an inline if/then/end

Comment (aljesusg) [reply app/helpers/manageiq/consumption/utils_helper.rb]: Fixed, Thanks @Fryguy

File: app/helpers/manageiq/consumption/time_converter_helper.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ module ManageIQ::Consumption
   module TimeConverterHelper
     VALID_INTERVAL_UNITS = %w(hourly daily weekly monthly yearly).freeze
 
-    def self.number_of_intervals(period:, interval:, calculation_date: Time.now, days_in_month: nil, days_in_year: nil)
+    def self.number_of_intervals(period:, interval:, calculation_date: Time.now.utc, days_in_month: nil, days_in_year: nil)
```
Comment (chargio) [inline app/helpers/manageiq/consumption/time_converter_helper.rb]: Time.current=?

File: app/helpers/manageiq/consumption/units_converter_helper.rb [hunk 3]
```diff
@@ -10,48 +10,48 @@ module ManageIQ::Consumption
...
-                                 value: 1_048_576},
-                      :'Gi' => { name: 'gibi',
-                                 value: 1_073_741_824},
-                      :'Ti' => { name: 'tebi',
-                                 value: 1_099_511_627_776},
-                      :'Pi' => { name: 'pebi',
-                                 value: 1_125_899_906_842_624},
-                      :'Ei' => { name: 'exbi',
-                                 value: 1_152_921_504_606_846_976} }.freeze
+    SI_PREFIX = {  ''  => { :name  => '',
```
Comment (chargio) [inline app/helpers/manageiq/consumption/units_converter_helper.rb:13]: As they are going to be there as units, they seem to be conceptually a symbol.....

File: app/models/manageiq/consumption/showback_charge.rb [hunk 4]
```diff
@@ -14,54 +16,80 @@ class ManageIQ::Consumption::ShowbackCharge < ApplicationRecord
   before_create :stored_data_event
   default_value_for :stored_data, {}
 
+  # Set teh cost to 0
```
Comment (chargio) [inline app/models/manageiq/consumption/showback_charge.rb]: Set the cost to 0 💃

File: app/helpers/manageiq/consumption/time_converter_helper.rb (hunk 2 above)
Comment (aljesusg) [reply app/helpers/manageiq/consumption/time_converter_helper.rb]: Rubocop offense when time.now... or time.zone.now or other...I set utc. Which is better?

Comment (aljesusg) [reply app/helpers/manageiq/consumption/time_converter_helper.rb]: Do not use Time.now without zone. Use one of Time.zone.now, Time.current, Time.now.in_time_zone, Time.now.utc, Time.now.getlocal, Time.now.iso8601, Time.now.jisx0301, Time.now.rfc3339, Time.now.to_i, Time.now.to_f instead

Time.current so?

File: app/helpers/manageiq/consumption/units_converter_helper.rb (hunk 3 above)
Comment (aljesusg) [reply app/helpers/manageiq/consumption/units_converter_helper.rb:13]: It's running without

File: app/models/manageiq/consumption/showback_charge.rb (hunk 4 above)
Comment (aljesusg) [reply app/models/manageiq/consumption/showback_charge.rb]: fixed

File: app/helpers/manageiq/consumption/time_converter_helper.rb (hunk 2 above)
Comment (chargio) [reply app/helpers/manageiq/consumption/time_converter_helper.rb]: @gtanzillo @lpichler In your experience, what is the best way to use this?
How it is done in the old code?
Should we merge this and open an issue?

**Review by chargio — APPROVED** [review]
Cleaning of code without changes. LGTM, the only comment is about time zones, as using current or utc would have implications

File: spec/helpers/time_converter_helper_spec.rb [hunk 5]
```diff
@@ -23,20 +23,23 @@
...
   context '#number of intervals on this month' do
-    let(:time_values) { [0.seconds, 15.minutes, 45.minutes, 1.hour, 90.minutes, 5.hours, 1.day, (1.5).days, 1.week, (1.4).weeks, 1.month - 1.second] }
+    let(:time_values) { [0.seconds, 15.minutes, 45.minutes, 1.hour, 90.minutes, 5.hours, 1.day, 1.5.days, 1.week, 1.4.weeks, 1.month - 1.second] }
 
     it 'minutely' do
       interval = 'minutely'
-      results = [1, 15, 45, 60, 90, 300, 24 * 60, 15 * 24 * 6, 7 * 24 * 60, 14 *  7 * 24 * 6, 24 * 60 * Time.days_in_month(Time.current.month)]
+      results = [1, 15, 45, 60, 90, 300, 24 * 60, 15 * 24 * 6, 7 * 24 * 60, 14 * 7 * 24 * 6, 24 * 60 * Time.days_in_month(Time.current.month)]
       expect(results.length).to eq(time_values.length)
       start_t = Time.current.beginning_of_month
```
Comment (chargio) [inline spec/helpers/time_converter_helper_spec.rb:32]: @Fryguy Should we use Time.current or time utc?

**Pull request merged**